File: demo_app/www/customer/customer.py
import frappe

@frappe.whitelist()
def save_customer(customer_name, email, address):
    try:
        doc = frappe.get_doc({
            "doctype": "Customer Info",  # <-- Doctype का सही नाम डालो
            "customer_name": customer_name,
            "email": email,
            "address": address
        })
        doc.insert(ignore_permissions=True)
        frappe.db.commit()
        return True
    except Exception as e:
        frappe.log_error(frappe.get_traceback(), "Customer Save Error")
        return False

File: demo_app/www/customer/customer.js
document.getElementById("customer-form").addEventListener("submit", function(e) {
    e.preventDefault();

    let customer_name = document.getElementById("customer_name").value;
    let email = document.getElementById("email").value;
    let address = document.getElementById("address").value;

    frappe.call({
        method: "demo_app.www.customer.customer.save_customer",  
        args: {
            customer_name: customer_name,
            email: email,
            address: address
        },
        headers: { "X-Frappe-CSRF-Token": frappe.csrf_token },
        callback: function(r) {
            let msg = document.getElementById("msg");
            if(r.message) {
                msg.innerText = "✅ Customer Saved Successfully!";
                msg.className = "text-green-600 text-lg";
                document.getElementById("customer-form").reset();
            } else {
                msg.innerText = "❌ Error saving customer!";
                msg.className = "text-red-600 text-lg";
            }
        }
    });
});
